pkg/sdk/roles.go

```diff
@@ -44,51 +48,115 @@ func (v *Role) ObjectType() ObjectType {
 	return ObjectTypeRole
 }
 
-// RoleCreateOptions contains options for creating a role.
-type RoleCreateOptions struct{}
+type roleDBRow struct {
+	CreatedOn       time.Time      `db:"created_on"`
+	Name            string         `db:"name"`
+	IsDefault       sql.NullString `db:"is_default"`
+	IsCurrent       sql.NullString `db:"is_current"`
+	IsInherited     sql.NullString `db:"is_inherited"`
+	AssignedToUsers int            `db:"assigned_to_users"`
+	GrantedToRoles  int            `db:"granted_to_roles"`
+	GrantedRoles    int            `db:"granted_roles"`
+	Owner           sql.NullString `db:"owner"`
+	Comment         sql.NullString `db:"comment"`
+}
 
-func (v *roles) Create(ctx context.Context, id AccountObjectIdentifier, opts *RoleCreateOptions) error {
-	sql := fmt.Sprintf(`CREATE ROLE %v`, id.FullyQualifiedName())
-	_, err := v.client.exec(ctx, sql)
-	return err
+func (row *roleDBRow) toRole() Role {
```

Comment:
please make it a `convert` method and add `var _ convertibleRow[Role] = new(roleDBRow)` (check more in database_role.go)